azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/collection/ResourcesInGroup.java

```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.resources.fluentcore.collection;
+
+public interface ResourcesInGroup<T, DefinitionT> extends
+    SupportsListing<T>,
+    SupportsCreating<DefinitionT>,
+    SupportsDeleting {}
```

Comment:
I see what you mean - from the REST yes everything is actually by ID. From the user's perspective, we can have two interfaces
`SupportsGetting` which gets by name
`SupportsGettingByGroup` which gets by group and name

The 1st one will be used by resource group collection and all other in group collections.
The 2nd one will be used by all other top level collections.